webui/react/src/pages/ExperimentDetails/CodeViewer/CodeViewer.tsx

```diff
@@ -342,33 +342,46 @@ const CodeViewer: React.FC<Props> = ({
                   */
                   <Tooltip title="Download File">
                     {
-                      !String(activeFile.key).includes('Configuration') && (
-                        // hiding the download for configs until next iteration
-                        <DownloadOutlined
-                          className={css.noBorderButton}
-                          onClick={(e) => {
-                            const filePath = String(activeFile.key);
-                            if (filePath.includes('Configuration')) {
-                              const url = filePath.includes('runtime')
-                                ? URL.createObjectURL(new Blob([ runtimeConfig ]))
-                                : URL.createObjectURL(new Blob([ submittedConfig as string ]));
-
-                              handlePath(e, {
-                                external: true,
-                                path: url,
-                              });
-                            } else {
-                              handlePath(e, {
-                                external: true,
-                                path: paths.experimentFileFromTree(
-                                  experimentId,
-                                  String(activeFile.key),
-                                ),
-                              });
-                            }
-                          }}
-                        />
-                      )
+                      <DownloadOutlined
+                        className={css.noBorderButton}
+                        onClick={(e) => {
+                          const filePath = String(activeFile.key);
+                          if (filePath.includes('Configuration')) {
+                            const isRuntimeConf = filePath.includes('runtime');
+                            const url = isRuntimeConf
+                              ? URL.createObjectURL(new Blob([ runtimeConfig ]))
+                              : URL.createObjectURL(new Blob([ submittedConfig as string ]));
+
+                            let timeout: NodeJS.Timeout;
+                            const anchor = document.createElement('a');
+                            const clickHandler = () => {
+                              timeout = setTimeout(() => {
+                                URL.revokeObjectURL(url);
+                              }, 300);
+                            };
```

Comment:
That would be nice if we can call `URL.revokeObjectURL(url);` after finishing download.

Reply:
You dont need to change it in this PR, and idk if its doable or not. but the thing in my mind was to create ObjectURL every time a file is selected, and revoke the previous one. Then we dont need to use timer I guess?